substrate/frame/collective/src/lib.rs

```diff
@@ -666,6 +680,26 @@ fn get_result_weight(result: DispatchResultWithPostInfo) -> Option<Weight> {
 }
 
 impl<T: Config<I>, I: 'static> Pallet<T, I> {
+	/// The active proposals.
+	#[cfg(any(feature = "std", feature = "runtime-benchmarks"))]
+	pub fn proposals() -> BoundedVec<PreimageHash, T::MaxProposals> {
+		Voting::<T, I>::iter()
+			.map(|(k, _v)| k)
+			.collect::<Vec<_>>()
+			.try_into()
+			.expect("proposals should always be less than MaxProposals; qed")
+	}
+
+	/// The bounded counterpart of an active proposal.
+	pub fn proposal_of(proposal_hash: &PreimageHash) -> Option<<T as Config<I>>::Proposal> {
+		match Voting::<T, I>::contains_key(proposal_hash) {
+			false => None,
+			true => T::Preimages::fetch(&proposal_hash, None)
+				.ok()
+				.and_then(|x| <T as Config<I>>::Proposal::decode(&mut &x[..]).ok()),
```

Comment:
```suggestion
			true => T::Preimages::fetch(&proposal_hash, None)
				.ok()
				.and_then(|x| <T as Config<I>>::Proposal::decode(&mut &x[..]).ok()),
```

should these `.ok()`s ever return `None`? If not, maybe best to log a warning in the instance that they do instead of silently returning `None`?